operator/operator_test.go

```diff
@@ -164,11 +164,17 @@ func initEnvironment(t *testing.T) *environment {
 }
 
 func (e *environment) WaitPhase() {
-	waitBlock := func() {
-		time.Sleep(blockTime * time.Second)
+	sub, err := e.operator.ethClient.SubscribeToBlocks()
```

Comment:
This blocks subscription is unnecessarily complex because it retrieves _all_ receipts of the whole block. Please just make it work with a simple header subscription, the only information you need is the block number.

Reply:
I tried it before with a header subscription, but it did not work. I can try it again if you think it is worth it.

Reply:
Ok it seems to work now, i probably did not have the `-1` there while trying it before.